### PR TITLE
Housekeeping/perform routine maintenance

### PR DESCRIPTION
**Pull request by pjvv**
Changes proposed in this pull request:

* updated README
* decreased the Docker image size
* moved to using `.flaskenv` file appropriately
* added lefthook
* updated dependencies

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -128,5 +128,5 @@ dmypy.json
 # Pyre type checker
 .pyre/
 
-# Visual studio
-.vscode/
```
Comment (emrojo) [inline .gitignore:132]: Is this not needed?

**Review by emrojo — DISMISSED** [review]

File: .flaskenv [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+# https://flask.palletsprojects.com/en/1.1.x/cli/#application-discovery
+FLASK_APP=baracoda
+
+# https://flask.palletsprojects.com/en/1.1.x/cli/#setting-command-options
+FLASK_RUN_HOST=0.0.0.0
+FLASK_RUN_PORT=8000
+
+# https://flask.palletsprojects.com/en/1.1.x/config/#environment-and-debug-features
+FLASK_ENV=development
```
[the remark this answers is not in the document]
Comment (pjvv) [reply .flaskenv:12]: Yeah, `.flaskenv` is only read when using `flask run`. On UAT and prod we run the server using gunicorn.

File: .github/workflows/automated_release_and_build.yml [hunk 3]
```diff
@@ -30,46 +28,53 @@ jobs:
...
+          docker.pkg.github.com/${IMAGE_NAME}:${GITHUB_REF##*/}
+          docker.pkg.github.com/${IMAGE_NAME}:${{ env.RELEASE_VERSION }}
+
+      - name: Push release tag image to registry
+        run: >-
+          docker push docker.pkg.github.com/${IMAGE_NAME}:${{ env.RELEASE_VERSION }}
+
+      - uses: actions/delete-package-versions@v1
+        with:
+          package-name: "${{ github.event.repository.name }}"
```
[the remark this answers is not in the document]
Comment (pjvv) [reply .github/workflows/automated_release_and_build.yml:80]: Yeah, the oldest one.

[the remark this answers is not in the document]
Comment (pjvv) [reply .github/workflows/automated_release_and_build.yml:80]: I've added a comment.

File: .github/workflows/check_release_name_bump.yml [hunk 4]
```diff
@@ -8,20 +9,31 @@ on:
...
     steps:
-    - uses: actions/checkout@v2
+      - uses: actions/checkout@v2
+
+      - name: Set excluded release types
+        run: >-
+          if [ "${{ github.base_ref }}" = "master" ]; then
+            echo "exclude_types=draft|prerelease" >> $GITHUB_ENV
+          elif [ "${{ github.base_ref }}" = "develop" ]; then
+            echo "exclude_types=draft" >> $GITHUB_ENV
```
[the remark this answers is not in the document]
Comment (pjvv) [reply .github/workflows/check_release_name_bump.yml:21]: Yeah they define it [here](https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable)

File: .gitignore (hunk 1 above)
Comment (pjvv) [reply .gitignore:132]: We can put recommended extensions and settings in this folder in the future.

File: .flaskenv (hunk 2 above)
[the remark this answers is not in the document]
Comment (emrojo) [reply .flaskenv:12]: But then don't we have the risk of using this if we ever replace current deployment? Like replacing gunicorn with something else? 

Shouldnt the contents of the file be moved in a different file, like .env.example? So we dont have that 'development' reference in production code?

**Review by emrojo — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (emrojo) [reply .gitignore:132]: Sounds good, so everybody will use same extensions.

File: .github/workflows/check_release_name_bump.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (emrojo) [reply .github/workflows/check_release_name_bump.yml:21]: Good, didnt know

File: .flaskenv (hunk 2 above)
[the remark this answers is not in the document]
Comment (pjvv) [reply .flaskenv:12]: Well if we ever run a production server using `flask run` (which they strongly discourage) then maybe we have bigger problems?! 😆

**Pull request merged**